pkg/apis/logging/v1/clusterlogging_types.go

```diff
@@ -210,6 +215,147 @@ type CuratorSpec struct {
 	Schedule string `json:"schedule"`
 }
 
+// OutputConfSpec represents the output configuration for tuning parameters per
+// output type, e.g. Fluentd. The tuning parameters are applied for all outputs
+// of the same type. Currently supported: `fluentd`.
+type OutputConfSpec struct {
+	Fluentd *FluentdOutputSpec `json:"fluentd,omitempty"`
+}
+
+// FluentdOutputSpec represents the configuration for outputs of type fluentd.
+// It supports overriding settings for fluentd buffer configurations. The settings
+// apply to all buffer configurations of each fluentd output.
+type FluentdOutputSpec struct {
+	Buffer *FluentdBufferSpec `json:"buffer,omitempty"`
+}
+
+const (
+	// ThrowExceptionAction raises an exception when output buffer is full
+	ThrowExceptionAction OverflowActionType = "throw_exception"
+	// BlockAction blocks processing inputs when output buffer is full
+	BlockAction OverflowActionType = "block"
+	// DropOldestChunkAction drops oldest chunk to accept newly incoming chunks
+	// when buffer is full
+	DropOldestChunkAction OverflowActionType = "drop_oldest_chunk"
+)
+
+type OverflowActionType string
+
+const (
+	// Flush one chunk per time key
+	FlushModeLazy FlushModeType = "lazy"
+	// Flush chunks per specified time via FlushInterval
+	FlushModeInterval FlushModeType = "interval"
+	// Flush immediately after events appended to chunks
+	FlushModeImmediate FlushModeType = "immediate"
+)
+
+type FlushModeType string
+
+const (
+	// RetryExponentialBackoff increases wait time exponentially between failures
+	RetryExponentialBackoff RetryTypeType = "exponential_backoff"
+	// RetryPeriodic to retry sending to output periodically on fixed intervals
+	RetryPeriodic RetryTypeType = "periodic"
+)
+
+type RetryTypeType string
+
+// FluentdBufferSpec represents a subset of fluentd buffer parameters to tune
+// the buffer configuration for all fluentd outputs. It supports a subset of
+// parameters to configure buffer and queue sizing, flush operations and retry
+// flushing.
+//
+// For general parameters refer to:
+// https://docs.fluentd.org/configuration/buffer-section#buffering-parameters
+//
+// For flush parameters refer to:
+// https://docs.fluentd.org/configuration/buffer-section#flushing-parameters
+//
+// For retry parameters refer to:
+// https://docs.fluentd.org/configuration/buffer-section#retries-parameters
+type FluentdBufferSpec struct {
+	// ChunkLimitSize represents the maximum size of each chunk. Events will be
+	// written into chunks until the size of chunks become this size. For data
+	// type pattern see:
+	// https://docs.fluentd.org/configuration/config-file#supported-data-types-for-values
+	//
+	// Notice: The OpenAPI validation pattern is an ECMA262 regular expression
+	// (See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#properties)
+	//
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([kmgtKMGT]{0,1})$"
+	ChunkLimitSize string `json:"chunkLimitSize"`
```

Comment:
these should be underscore instead of camelcase per the enhancement spec: 
```The field names are copied exactly from fluentd 1.0, including the use of "_" rather than camelCase. This emphasizes that these settings are not a general API, they set defaults directly for the underlying fluentd.```